docs/source/tutorial/trainer.rst

```diff
@@ -0,0 +1,308 @@
+
+1. Retrieve the data from dataset and construct a mini-batch (:class:`~chainer.dataset.Iterator`)
+2. Pass the mini-batch to the model and calculate the loss
+3. Update the parameters of the model (:class:`~chainer.Optimizer`)
+
+Now let's create the :class:`~chainer.training.Updater` object !
+
+.. testsetup:: *
+
+    from chainer.datasets import mnist
+
+    class MLP(Chain):
+
+        def __init__(self, n_mid_units=100, n_out=10):
+            super(MLP, self).__init__()
+            with self.init_scope():
+                self.l1 = L.Linear(None, n_mid_units)
+                self.l2 = L.Linear(None, n_mid_units)
+                self.l3 = L.Linear(None, n_out)
+
+        def __call__(self, x):
+            h1 = F.relu(self.l1(x))
+            h2 = F.relu(self.l2(h1))
+            return self.l3(h2)
+
+    model = MLP()
+
+    batchsize = 128
+
+    train, test = mnist.get_mnist()
+    train_iter = iterators.SerialIterator(train, batchsize)
+    test_iter = iterators.SerialIterator(test, batchsize, False, False)
+
+.. testcode::
+
+    max_epoch = 10
+
+    # Wrapp your model by Classifier and include the process of loss calculation within your model.
+    # Since we do not specify a loss funciton here, the default 'softmax_cross_entropy' is used.
+    model = L.Classifier(model)
+
+    # selection of your optimizing method
+    optimizer = optimizers.MomentumSGD()
+
+    # Give the optimizer a reference to the model
+    optimizer.setup(model)
+
+    # Get an updater that uses the Iterator and Optimizer
+    updater = training.StandardUpdater(train_iter, optimizer)
+
+.. note::
+
+    Here, the model defined above is passed to :class:`~chainer.links.Classifier` and changed to a new :class:`~chainer.Chain`. :class:`~chainer.links.Classifier`, which in fact inherits from the :class:`~chainer.Chain` class, keeps the given :class:`~chainer.Chain` model in its :attr:`~chainer.links.Classifier.predictor` attribute. Once you give the input data and the corresponding class labels to the model by the ``()`` operator,
+
+    1. :meth:`~chainer.links.Classifier.__call__` of the model is invoked. The data is then given to :attr:`~chainer.links.Classifier.predictor` to obtain the output ``y``.
+    2. Next, together with the given labels, the output ``y`` is passed to the loss function which is determined by :attr:`~chainer.links.Classifier.lossfun` argument in the constructor of :class:`~chainer.links.Classifier`.
+    3. The loss is returned as a :class:`~chainer.Variable`.
+
+    In :class:`~chainer.links.Classifier`, the :attr:`~chainer.links.Classifier.lossfun` is set to
+    :meth:`~chainer.functions.softmax_cross_entropy` as default.
+
+    :class:`~chainer.training.StandardUpdater` is the simplest class among several updaters. There are also the :class:`~chainer.training.ParallelUpdater` and the :class:`~chainer.training.updaters.MultiprocessParallelUpdater` to utilize multiple GPUs.
```

Comment:
Can you include what the difference is between ParallelUpdater and MultiprocessParallelUpdater here?